stock_change_qty_reason/models/stock_inventory_line.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2016-2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockInventoryLine(models.Model):
+    """Class to inherit model stock.inventory.line"""
+    _inherit = "stock.inventory.line"
+
+    def _get_move_values(self, qty, location_id, location_dest_id, out):
```

Comment:
define a field 'reason' in stock.inventory.line, and fetch the value, to then send it to the stock move.